From 2a11ba7d7d21b80cd1d7a362ca2ca401fc3b34dd Mon Sep 17 00:00:00 2001
From: Eric Yang <eric.yang2@amd.com>
Date: Wed, 23 Mar 2016 18:49:12 -0400
Subject: [PATCH 0985/1110] drm/amd/dal: fix DP active dongle downstream
 hotplug

handle_hpd_rx_irq need the same headless fix as handle_hpd_irq

Signed-off-by: Eric Yang <eric.yang2@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
---
 drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c
index 5b92771..038dea4 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm.c
@@ -730,6 +730,11 @@ static void handle_hpd_rx_irq(void *param)
 		/* Downstream Port status changed. */
 		if (dc_link_detect(aconnector->dc_link, false)) {
 			amdgpu_dm_update_connector_after_detect(aconnector);
+
+			drm_modeset_lock_all(dev);
+			dm_restore_drm_connector_state(dev, connector);
+			drm_modeset_unlock_all(dev);
+
 			drm_kms_helper_hotplug_event(dev);
 		}
 	}
-- 
2.7.4

